pkg/rpcservice/middleware.go

```diff
@@ -29,6 +48,41 @@ func authorize(ctx context.Context) error {
 	return nil
 }
 
+// Populate the context with:
+// 1. The telemetry client from the RPC Service
+// 2. The event metadata
+func updateContextWithTelemetry(ctx context.Context, methodName string, server *RPCService) context.Context {
+	// If the context is nil for whatever reason, create an empty one
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	// if getting the config errors, don't fail running the command
+	merchant, _ := server.cfg.UserCfg.Profile.GetAccountID()
+	useragent := getUserAgentFromGrpcMetadata(ctx)
+
+	telemetryMetadata := stripe.NewEventMetadata()
+	telemetryMetadata.SetMerchant(merchant)
+	telemetryMetadata.SetCommandPath(methodName)
+	telemetryMetadata.SetUserAgent(useragent)
+
+	newCtx := stripe.WithEventMetadata(stripe.WithTelemetryClient(ctx, server.TelemetryClient), telemetryMetadata)
+	return newCtx
+}
+
+func getUserAgentFromGrpcMetadata(ctx context.Context) string {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return ""
+	}
+
+	if len(md["user-agent"]) != 1 {
+		return ""
+	}
+
+	return md["user-agent"][0]
```

Comment:
I've seen people `strings.Join` the array of user agents, so I'm wondering if we need to do that. I *think* if your client sets a custom user agent, there could be multiple ones: the default one from the gRPC library, and your custom one. But I'm not sure.

Reply:
I need to mess around with sending a custom agent from the client but I assumed it would be something like this since the user_agent is a list:
```
"user_agent":["grpc-node-js/1.3.7", "vscode/2.6.1"]
```

But if people use a strings.Join, it would look something like this? 
```
"user_agent":["grpc-node-js/1.3.7, vscode/2.6.1"]
```
For the former, I need to mess around with it to see if there is an order to these (i.e. the default one always comes first so we'd want the lastest one). For the latter, I'm not quite sure what we'd do there. Let me look into this a bit more.

Reply:
Oh sorry, I meant that we might need to handle the case where the client sends an array of multiple user agents like `"user_agent":["grpc-node-js/1.3.7", "vscode/2.6.1"]`. In that case, I've seen other gRPC server code join the strings to together into `"grpc-node-js/1.3.7, vscode/2.6.1"` or something, so I was wondering if we'll want our interceptor to do the same. If we join them, I don't think order will matter.

Reply:
Ohh, I see what you mean. Yeah I like the idea of including them all. We can choose to filter in them in the hubble dashboard if we want to